vulnerability/events_creator.go

```diff
@@ -0,0 +1,152 @@
+	Severity       string               `json:"severity,omitempty"`
+	Classification string               `json:"classification,omitempty"`
+	PublishedDate  *time.Time           `json:"published_date,omitempty"`
+}
+
+type Package struct {
+	FixedVersion string `json:"fixed_version,omitempty"`
+	Version      string `json:"version,omitempty"`
+	Name         string `json:"name,omitempty"`
+}
+
+type Scanner struct {
+	Version string `json:"version,omitempty"`
+	Vendor  string `json:"vendor,omitempty"`
+}
+
+type Score struct {
+	Impact  float64 `json:"impact,omitempty"`
+	Base    float64 `json:"base,omitempty"`
+	Version string  `json:"version,omitempty"`
+}
+
+type Resource struct {
+	ID   string
+	Name string
+}
+
+const (
+	vulScannerVersion     = "v0.35.0"
+	vulScannerVendor      = "Trivy"
+	vulScoreSystemClass   = "CVSS"
+	vulScoreSource        = "nvd"
+	vulScoreSourceVersion = "3.0"
+	vulEcsCategory        = "vulnerability"
+	vulIndex              = "logs-cloud_security_posture.vulnerabilities-default"
+)
+
+func createVulnerabilityEvent(reportResult trivy_types.Result, vul trivy_types.DetectedVulnerability, ins ec2.Ec2Instance) beat.Event {
+	timestamp := time.Now()
+	cvssScore := getCVSSScore(vul, vulScoreSource)
+	return beat.Event{
+		// TODO: Maybe configure or get from somewhere else?
+		Meta:      mapstr.M{libevents.FieldMetaIndex: vulIndex},
+		Timestamp: timestamp.UTC(),
+		Fields: mapstr.M{
+			"cloudbeat": version.CloudbeatVersion(),
+			// TODO: Do we need sequence?
+			"event": transformer.BuildECSEvent(timestamp.Unix(), timestamp.UTC(), []string{vulEcsCategory}),
+			"resource": Resource{
+				ID:   ins.GetResourceId(),
+				Name: ins.GetResourceName(),
+			},
+			"finding": Finding{
+				Target: reportResult.Target,
+				Class:  reportResult.Class,
+				Type:   reportResult.Type,
+				Vul: Vulnerability{
+					Cvss:       vul.CVSS,
+					DataSource: vul.DataSource,
+					Scanner: Scanner{
+						// TODO: Populate with what?
+						Version: vulScannerVersion,
+						Vendor:  vulScannerVendor,
+					},
+					Score: Score{
+						Impact:  cvssScore,
+						Base:    cvssScore,
+						Version: vulScoreSourceVersion,
+					},
+					Package: Package{
+						FixedVersion: vul.FixedVersion,
+						Name:         vul.PkgName,
+						Version:      vul.InstalledVersion,
+					},
+					Cwe:            vul.CweIDs,
+					ID:             vul.VulnerabilityID,
+					Title:          vul.Title,
+					Enumeration:    getIdentifierType(vul.VulnerabilityID),
+					Reference:      vul.PrimaryURL,
+					Description:    vul.Description,
+					Severity:       vul.Severity,
+					Classification: vulScoreSystemClass,
+					PublishedDate:  vul.PublishedDate,
+				},
+			},
+		},
+	}
+}
+
+func getIdentifierType(id string) string {
+	return strings.Split(id, "-")[0]
+}
+
+func getCVSSScore(vul trivy_types.DetectedVulnerability, source db_types.SourceID) float64 {
+	if cvss, ok := vul.CVSS[source]; ok {
+		return cvss.V3Score
+	}
+
+	return 0
```

Comment:
is this a valid score?

Reply:
Looking at:
- https://www.first.org/cvss/v3.0/specification-document

I can see the following:
<img width="1196" alt="Screen Shot 2023-04-10 at 16 17 10" src="https://user-images.githubusercontent.com/12246907/230908405-a6d269e0-cb86-4372-a3ed-0e178eb74e10.png">

Do you think that I should look at some other place?